subsys/bluetooth/host/gatt.c

```diff
@@ -1478,8 +1477,16 @@ ssize_t bt_gatt_attr_write_ccc(struct bt_conn *conn,
 	}
 
 	/* Confirm write if cfg is managed by application */
-	if (ccc->cfg_write && !ccc->cfg_write(conn, attr, value)) {
-		return BT_GATT_ERR(BT_ATT_ERR_WRITE_NOT_PERMITTED);
+	if (ccc->cfg_write) {
+		ssize_t write = ccc->cfg_write(conn, attr, value);
+
+		if (write < 0) {
+			return write;
+		}
+
+		if (write != sizeof(value)) {
```

Comment:
In my opinion this change breaks the API. Previously true (1) meant success and false (0) meant error. Now 0, 1 means error and 2 means success.

If we are fine with this API break we can leave it.
If we are not fine with this API break we should accept both 1 and 2 as success value.

@jhedberg Do we accept to have this API change?

Reply:
Accepting return values of both 1 and 2 doesn't just sound like a good thing for backwards compatibility - it also makes sense from the spec perspective, i.e. it could simply mean that callback only consumed the least significant byte of the value since only that one has the two bits that the current core spec assigns meaning to.

Reply:
@jhedberg I will change it to only accept 2 bytes.